lib/imageflow/graph.ex

```diff
@@ -120,6 +132,37 @@ defmodule Imageflow.Graph do
     |> append_node(%{encode: %{io_id: io_id, preset: preset_for(encoder, opts)}})
   end
 
+  @doc """
+  Returns the converted image as a string.
+
+  No further processing operations should be appended at the current branch after this call.
+
+  The last two arguments specify the encoder and optional encoding parameters.
+
+  The following parameters are valid encoders:
+  * `:jpg`: Alias to `:mozjpeg`
+  * `:jpeg`: Alias to `:mozjpeg`
+  * `:png`: Alias to `:lodepng`
+  * `:webp: Alias to `:webplossless
+  * `:mozjpeg`
+  * `:gif`
+  * `:lodepng`: Lossless PNG
+  * `:pngquant`: Lossy PNG
+  * `:webplossy`: Lossy WebP
+  * `:webplossless`: Lossless WebP
+
+  Check the official [encoding documentation](https://docs.imageflow.io/json/encode.html) to see the parameters available to each encoder
+  """
+
+  @spec encode_to_string(t, binary | atom, map) :: t
+  def encode_to_string(%{io_count: io_count} = graph, encoder \\ :png, opts \\ %{}) do
+    io_id = io_count + 1
+
+    graph
+    |> add_output(io_id, :bytes)
+    |> append_node(%{encode: %{io_id: io_id, preset: preset_for(encoder, opts)}})
+  end
```

Comment:
it seems this is mostly duplicate code from `encode_to_file/4`, correct? How do you feel about refactoring to extract the duplication to a separate function?